src/server/pps/server/api_server.go

```diff
@@ -2135,6 +2135,29 @@ func (a *apiServer) CreatePipelineV2(ctx context.Context, request *pps.CreatePip
 	}, nil
 }
 
+func (a *apiServer) RerunPipeline(ctx context.Context, request *pps.RerunPipelineRequest) (response *emptypb.Empty, err error) {
+	metricsFn := metrics.ReportUserAction(ctx, a.reporter, "RerunPipeline")
+	defer func(start time.Time) { metricsFn(start, err) }(time.Now())
+
+	info, err := a.inspectPipeline(ctx, request.Pipeline, true)
+
+	if err != nil {
+		return nil, err
```

Comment:
Wrap errors.  `return errors.Wrap(err, "inspect")`.  errors is src/internal/errors.